### PR TITLE
fix: fix sonar blocker and critical bug

### PR DESCRIPTION
**Pull request by mnianqi**

### CODE REVIEW

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **42.10526315789474**
- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **2**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=fix-develop-sonar-bug&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **0.0**
- Status: **ERROR**, MetricKey: **new_critical_violations**, ActualValue: **1**
- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **1**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=fix-develop-sonar-bug&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **0.0**
- Status: **ERROR**, MetricKey: **new_critical_violations**, ActualValue: **1**
- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **1**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=fix-develop-sonar-bug&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **0.0**
- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **2**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=fix-develop-sonar-bug&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **0.0**
- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **1**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=fix-develop-sonar-bug&id=tapdata

File: iengine/iengine-app/src/main/java/io/tapdata/flow/engine/V2/node/hazelcast/data/HazelcastSchemaTargetNode.java [hunk 1]
```diff
@@ -147,8 +147,8 @@ protected void doInit(@NotNull Context context) throws TapCodeException {
 							if (engine instanceof Closeable) {
 								((Closeable) engine).close();
 							}
-						} catch (Exception e) {
```
Comment (ply0011) [inline iengine/iengine-app/src/main/java/io/tapdata/flow/engine/V2/node/hazelcast/data/HazelcastSchemaTargetNode.java:150]: 改为warn级别的日志输出

File: iengine/iengine-app/src/main/java/io/tapdata/task/TapdataTaskScheduler.java [hunk 2]
```diff
@@ -280,6 +280,9 @@ public static void main(String[] args) throws InterruptedException {
 		while (true) {
 			System.out.println(schedule.isDone());
 			Thread.sleep(1000);
+			if (Thread.currentThread().isInterrupted()){
```
Comment (ply0011) [inline iengine/iengine-app/src/main/java/io/tapdata/task/TapdataTaskScheduler.java:283]: 这个main方法删除

File: iengine/modules/proxy-client-module/src/main/java/io/tapdata/proxy/client/ProxySubscriptionManager.java [hunk 3]
```diff
@@ -91,33 +91,29 @@ public ProxySubscriptionManager() {
 			maxFrequencyLimiter = new MaxFrequencyLimiter(500, this::syncSubscribeIds);
 		}
 	}
-	public void startIMClient(List<String> baseURLs, String accessToken) {
+	public synchronized void startIMClient(List<String> baseURLs, String accessToken) {
```
Comment (ply0011) [inline iengine/modules/proxy-client-module/src/main/java/io/tapdata/proxy/client/ProxySubscriptionManager.java:94]: 最好别在方法上锁

File: iengine/iengine-app/src/main/java/io/tapdata/inspect/InspectFactory.java [hunk 4]
```diff
@@ -30,24 +30,20 @@ public class InspectFactory {
     private static InspectService inspectService;
 
 
-    public static InspectService getInstance(ClientMongoOperator clientMongoOperator, SettingService settingService) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+    public static synchronized InspectService getInstance(ClientMongoOperator clientMongoOperator, SettingService settingService) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
```
Comment (ply0011) [inline iengine/iengine-app/src/main/java/io/tapdata/inspect/InspectFactory.java:33]: 最好别在方法上锁

**Review by ply0011 — CHANGES_REQUESTED** [review]

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **0.0**
- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **1**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=fix-develop-sonar-bug&id=tapdata

**Review by ply0011 — APPROVED** [review]

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **0.0**
- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **1**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=fix-develop-sonar-bug&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **0.0**
- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **1**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=fix-develop-sonar-bug&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **0.0**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=fix-develop-sonar-bug&id=tapdata

**Comment by ply0011** [comment]
# exception case

This is fix issues in the old code identified by the Sonar scan. It is agreed that unit tests are not required

**Pull request merged**